Update file sync planning with common sync scenarios and solutions (#100991)
Azure/azure-content: includes/storage-files-migration-namespace-mapping.md

It's possible that, in your situation, a set of folders can logically sync to the same Azure file share (by using the new common root folder approach mentioned earlier). But it might still be better to regroup folders so they sync to two instead of one Azure file share. You can use this approach to keep the number of files and folders per file share balanced across the server. You can also split your on-premises shares and sync across more on-premises servers, adding the ability to sync with 30 more Azure file shares per extra server.

#### Common File Sync Scenarios and Considerations

| # | Sync Scenarios | Supported | Considerations (or limitations) | Solution (or workaround) | 
|---|---|:---:|---|---|

> | 1 | File Server with multiple disks/volumes and multiple shares to same target Azure File share (Consolidation) | No | A target Azure file share (cloud endpoint) only supports syncing with 1 sync group. <br/> <br/> A Sync group only support 1 server endpoint per registered server. | 1) Start with syncing 1 disk (it's root volume) to target Azure file share. Starting with largest disk/volume will help with storage requirements on-prem. Config cloud tiering to tier all data to cloud, thereby freeing up space on the file server disk. Move data from other volumes/shares into the current volume which is syncing. Continue the steps one by one until all data is tiered up to cloud/migrated.<br/> 2) Target 1 root volume (disk) at a time. Use cloud tiering to tier all data to target Azure file share. Remove server endpoint from sync group, re-create the endpoint with the next root volume/disk, sync and repeat the process. Note: Agent re-install might be required.<br/> 3) Recommend using multiple target Azure File Share (same or different storage account based on perf requirements) |

Reviewer: ```suggestion
| 1 | File Server with multiple disks/volumes and multiple shares to same target Azure file share (consolidation) | No | A target Azure file share (cloud endpoint) only supports syncing with one sync group. <br/> <br/> A sync group only supports one server endpoint per registered server. | 1) Start with syncing one disk (its root volume) to target Azure file share. Starting with largest disk/volume will help with storage requirements on-premises. Configure cloud tiering to tier all data to cloud, thereby freeing up space on the file server disk. Move data from other volumes/shares into the current volume which is syncing. Continue the steps one by one until all data is tiered up to cloud/migrated.<br/> 2) Target one root volume (disk) at a time. Use cloud tiering to tier all data to target Azure file share. Remove server endpoint from sync group, re-create the endpoint with the next root volume/disk, sync and repeat the process. Note: Agent re-install might be required.<br/> 3) Recommend using multiple target Azure file shares (same or different storage account based on performance requirements) |
```

> | 1 | File Server with multiple disks/volumes and multiple shares to same target Azure File share (Consolidation) | No | A target Azure file share (cloud endpoint) only supports syncing with 1 sync group. <br/> <br/> A Sync group only support 1 server endpoint per registered server. | 1) Start with syncing 1 disk (it's root volume) to target Azure file share. Starting with largest disk/volume will help with storage requirements on-prem. Config cloud tiering to tier all data to cloud, thereby freeing up space on the file server disk. Move data from other volumes/shares into the current volume which is syncing. Continue the steps one by one until all data is tiered up to cloud/migrated.<br/> 2) Target 1 root volume (disk) at a time. Use cloud tiering to tier all data to target Azure file share. Remove server endpoint from sync group, re-create the endpoint with the next root volume/disk, sync and repeat the process. Note: Agent re-install might be required.<br/> 3) Recommend using multiple target Azure File Share (same or different storage account based on perf requirements) |

Reviewer: ```suggestion
| 1 | File server with multiple disks/volumes and multiple shares to same target Azure File share (Consolidation) | No | A target Azure file share (cloud endpoint) only supports syncing with 1 sync group. <br/> <br/> A Sync group only support 1 server endpoint per registered server. | 1) Start with syncing 1 disk (it's root volume) to target Azure file share. Starting with largest disk/volume will help with storage requirements on-prem. Config cloud tiering to tier all data to cloud, thereby freeing up space on the file server disk. Move data from other volumes/shares into the current volume which is syncing. Continue the steps one by one until all data is tiered up to cloud/migrated.<br/> 2) Target 1 root volume (disk) at a time. Use cloud tiering to tier all data to target Azure file share. Remove server endpoint from sync group, re-create the endpoint with the next root volume/disk, sync and repeat the process. Note: Agent re-install might be required.<br/> 3) Recommend using multiple target Azure File Share (same or different storage account based on perf requirements) |
```

> | 1 | File Server with multiple disks/volumes and multiple shares to same target Azure file share (consolidation) | No | A target Azure file share (cloud endpoint) only supports syncing with one sync group. <br/> <br/> A sync group only supports one server endpoint per registered server. | 1) Start with syncing one disk (its root volume) to target Azure file share. Starting with largest disk/volume will help with storage requirements on-premises. Configure cloud tiering to tier all data to cloud, thereby freeing up space on the file server disk. Move data from other volumes/shares into the current volume which is syncing. Continue the steps one by one until all data is tiered up to cloud/migrated.<br/> 2) Target one root volume (disk) at a time. Use cloud tiering to tier all data to target Azure file share. Remove server endpoint from sync group, re-create the endpoint with the next root volume/disk, sync and repeat the process. Note: Agent re-install might be required.<br/> 3) Recommend using multiple target Azure file shares (same or different storage account based on performance requirements) |

Reviewer: ```suggestion
| 1 | File server with multiple disks/volumes and multiple shares to same target Azure file share (consolidation) | No | A target Azure file share (cloud endpoint) only supports syncing with one sync group. <br/> <br/> A sync group only supports one server endpoint per registered server. | 1) Start with syncing one disk (its root volume) to target Azure file share. Starting with largest disk/volume will help with storage requirements on-premises. Configure cloud tiering to tier all data to cloud, thereby freeing up space on the file server disk. Move data from other volumes/shares into the current volume which is syncing. Continue the steps one by one until all data is tiered up to cloud/migrated.<br/> 2) Target one root volume (disk) at a time. Use cloud tiering to tier all data to target Azure file share. Remove server endpoint from sync group, re-create the endpoint with the next root volume/disk, sync, and repeat the process. Note: Agent re-install might be required.<br/> 3) Recommend using multiple target Azure file shares (same or different storage account based on performance requirements) |
```

> | 2 | File Server with single volume and multiple shares to same target Azure File share (Consolidation) | Yes* | Cannot have multiple server endpoints per registered server syncing to same target Azure file share (same as above) | Sync root* of the volume holding multiple shares or top-level folders. Refer [Share grouping concept](#share-grouping) and [Volume sync](#volume-sync) for more info. |

Reviewer: ```suggestion
| 2 | File server with single volume and multiple shares to same target Azure file share (consolidation) | Yes | Can't have multiple server endpoints per registered server syncing to same target Azure file share (same as above) | Sync root of the volume holding multiple shares or top-level folders. Refer to [Share grouping concept](#share-grouping) and [Volume sync](#volume-sync) for more information. |
```

> | 3 | File Server with multiple shares and/or volumes to multiple Azure file shares under single Storage Account (1:1 share mapping) | Yes | A single Windows Server instance (or cluster) can sync up to 30 Azure file shares.<br/><br/> A storage account is a scale target for performance. IOPS and throughput get shared across file shares.<br/><br/> Keep no. of items per sync group within 100 million items (files and folders) per share. Ideally best if it's below 20 or 30 million per share. | 1) Use multiple sync groups (no. of sync groups = no. of Azure file shares to sync to).<br/>  2) Only 30 shares can be synced in this scenario at a time. If you have more than 30 shares on that File Server, use [Share grouping concept](#share-grouping) and [Volume sync](#volume-sync) to reduce no. of root or top-level folders at source.<br/> 3) Use additional File Sync servers on-prem and split/move data to these servers to work around limitations on the source Windows server. |

Reviewer: ```suggestion
| 3 | File server with multiple shares and/or volumes to multiple Azure file shares under single storage account (1:1 share mapping) | Yes | A single Windows Server instance (or cluster) can sync up to 30 Azure file shares.<br/><br/> A storage account is a scale target for performance. IOPS and throughput get shared across file shares.<br/><br/> Keep number of items per sync group within 100 million items (files and folders) per share. Ideally it's best to stay below 20 or 30 million per share. | 1) Use multiple sync groups (number of sync groups = number of Azure file shares to sync to).<br/>  2) Only 30 shares can be synced in this scenario at a time. If you have more than 30 shares on that file server, use [Share grouping concept](#share-grouping) and [Volume sync](#volume-sync) to reduce the number of root or top-level folders at source.<br/> 3) Use additional File Sync servers on-premises and split/move data to these servers to work around limitations on the source Windows server. |
```

> | 4 | File Server with multiple shares and/or volumes to multiple Azure file shares under different Storage Account (1:1 share mapping) | Yes | A single Windows Server instance (or cluster) can sync up to 30 Azure file shares (same or different Storage account).<br/><br/> Keep no. of items per sync group within 100 million items (files and folders) per share. Ideally best if it's below 20 or 30 million per share. | *Same approach as above* |

Reviewer: ```suggestion
| 4 | File server with multiple shares and/or volumes to multiple Azure file shares under different storage account (1:1 share mapping) | Yes | A single Windows Server instance (or cluster) can sync up to 30 Azure file shares (same or different storage account).<br/><br/> Keep number of items per sync group within 100 million items (files and folders) per share. Ideally it's best to stay below 20 or 30 million per share. | Same approach as above |
```

> | 5 | Multiple File Servers with single (root volume or share) to same target Azure File share (Consolidation) | No | A sync group cannot use cloud endpoint (Azure file share) already configured in another sync group.<br/><br/> Though a sync group can have server endpoints on different File Servers, the files cannot be distinct. | *Follow guidance in Scenario # 1 above with additional consideration of targeting one file server at a time.* |

Reviewer: ```suggestion
| 5 | Multiple file servers with single (root volume or share) to same target Azure file share (consolidation) | No | A sync group can't use cloud endpoint (Azure file share) already configured in another sync group.<br/><br/> Although a sync group can have server endpoints on different file servers, the files can't be distinct. | *Follow guidance in Scenario # 1 above with additional consideration of targeting one file server at a time.* |
```
